app/models/item.rb

```diff
@@ -1,3 +1,9 @@
 class Item < ApplicationRecord
   belongs_to :supermarket
+  has_many :customer_items
+  has_many :customers, through: :customer_items
+
+  def name_supermarket
+    supermarket.name
```

Comment:
Nice! It's also okay to use `supermarket.name` in a view. It is an attribute of an association, which is fine to call in the view! We just don't want to call ActiveRecord calculations or queries in the view (i.e. you don't want .where, .count, .order, etc in the view)